pkg/distributor/distributor.go

```diff
@@ -1246,6 +1269,19 @@ func (d *Distributor) prePushValidationMiddleware(next PushFunc) PushFunc {
 			validatedExemplars += len(ts.Exemplars)
 		}
 
+		if len(dedupedPerMetric) > 0 {
+			// Emit tracing span events for metrics with deduped samples.
+			spanLogger, _ := spanlogger.NewWithLogger(ctx, d.log, "Distributor.prePushValidationMiddleware")
```

Comment:
I'm a little hesitant about adding a whole new span just to report some debug information, but I guess we can go with it and see if the cost is noticeable.